src/backend/distributed/sql/udfs/citus_drop_trigger/latest.sql

```diff
@@ -14,16 +9,28 @@ DECLARE
     sequence_names text[] := '{}';
     table_colocation_id integer;
     propagate_drop boolean := false;
+    drop_shards_metadata_only boolean := false;
 BEGIN
     FOR v_obj IN SELECT * FROM pg_event_trigger_dropped_objects()
                  WHERE object_type IN ('table', 'foreign table')
     LOOP
+        -- If both original and normal values are false, the table was a partition
+        -- its shards will be dropped when dropping the parent's shards, so we can skip
+        -- NOTE: the other way around is not true:
+        -- the table being a partition doesn't imply both original and normal values are false
+        IF v_obj.original = false AND v_obj.normal = false
```

Comment:
does this affect other cases (e.g. DROP SCHEMA)? does that matter?

Reply:
Yes, `DROP SCHEMA CASCADE` matters. I have tested the cases in the description and this `if` condition is still safe.